cookbook/session/avoid_session_start.rst

```diff
@@ -0,0 +1,54 @@
+.. index::
+    single: Sessions, cookies
+
+Avoid Starting Sessions for Anonymous Users
+===========================================
+
+Sessions in Symfony applications are automatically started whenever they are necessary.
+This includes writing in the user's session, creating a flash message and logging
+in users. In order to start the session, Symfony creates a cookie which will be
+added to every user request.
+
+However, there are other scenarios when a session is started automatically and a
+cookie will be created even for anonymous users. First, consider the following
+template code commonly used to display flash messages:
+
+.. code-block:: html+jinja
+
+    {% for flashMessage in app.session.flashbag.get('notice') %}
+        <div class="flash-notice">
+            {{ flashMessage }}
+        </div>
+    {% endfor %}
+
+Even if the user is not logged in and even if you haven't created any flash message,
+just calling the ``get()`` method of the ``flashbag`` will start a session. This
+may hurt your application performance because all users will receive a session
+cookie. To avoid this behavior, add a check before trying to access the flash messages:
+
+.. code-block:: html+jinja
+
+    {% if app.session.started %}
+        {% for flashMessage in app.session.flashbag.get('notice') %}
```

Comment:
Ok, so I don't think this is right - it's a confusing thing:

A) `Session::isStarted()` checks if the session has been started on *this* request or not. This is *not* what we want

B) `Request::hasPreviousSession()` checks if the user has a session cookie - i.e. if a session was created on a previous request. This is what we *do* want.

There's a huge confusion over this - I was just reading symfony/symfony#6036 and symfony/symfony#6388 about this - the functionality for this is not right in the core, which is why this is so difficult.